app/controllers/application_controller/queries.rb

```diff
@@ -0,0 +1,437 @@
+  #
+  #  While browsing like this, the user may want to divert temporarily to add a
+  #  comment or propose a name or something.  These actions are responsible for
+  #  keeping track of these search parameters, and eventually passing them back
+  #  to the show_object page.  Usually they just pass the query parameter
+  #  through via +pass_query_params+.
+  #
+  #  See Query and AbstractQuery for more detail.
+  #
+  ##############################################################################
+
+  # This clears the search/index saved in the session.
+  def clear_query_in_session
+    session[:checklist_source] = nil
+  end
+
+  # This stores the latest search/index used for use by create_species_list.
+  # (Stores the Query id in <tt>session[:checklist_source]</tt>.)
+  def store_query_in_session(query)
+    query.save unless query.id
+    session[:checklist_source] = query.id
+  end
+
+  # Get Query last stored on the "clipboard" (session).
+  def query_from_session
+    return unless (id = session[:checklist_source])
+
+    Query.safe_find(id)
+  end
+  # helper_method :query_from_session
+
+  # Get instance of Query which is being passed to subsequent pages.
+  def passed_query
+    Query.safe_find(query_params[:q].to_s.dealphabetize)
+  end
+  # helper_method :passed_query
+
+  # NOTE: If we're going to cache user stuff that depends on their present q,
+  # we'll need a helper to make the current QueryRecord (not just the id)
+  # available to templates as an ApplicationController ivar. Something like:
+  #
+  # def current_query_record
+  #   current_query = passed_query || query_from_session # could both be nil!
+  #   current_query_record = current_query&.record || "no_query"
+  # end
+
+  # Return query parameter(s) necessary to pass query information along to
+  # the next request. *NOTE*: This method is available to views.
+  def query_params(query = nil)
+    if browser.bot?
+      {}
+    elsif query
+      query.save unless query.id
+      { q: query.id.alphabetize }
+    else
+      @query_params || {}
+    end
+  end
+  # helper_method :query_params
+
+  def add_query_param(params, query = nil)
+    return params if browser.bot?
+
+    query_param = get_query_param(query)
+    if params.is_a?(String) # i.e., if params is a path
+      append_query_param_to_path(params, query_param)
+    else
+      params[:q] = query_param if query_param
+      params
+    end
+  end
+  # helper_method :add_query_param
+
+  def append_query_param_to_path(path, query_param)
+    return path unless query_param
+
+    if path.include?("?") # Does path already have a query string?
+      "#{path}&q=#{query_param}" # add query_param to existing query string
+    else
+      "#{path}?q=#{query_param}" # create a query string comprising query_param
+    end
+  end
+
+  # Allows us to add query to a path helper:
+  #   object_path(@object, q: get_query_param)
+  def get_query_param(query = nil)
```

Comment:
Outside the scope of this PR, 
but how about (globally) renaming this to something like
`query_id_string` or `query_id_alphabetized` or `q_param`?
("param" is overloaded, "get" is redundant)